src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -8027,4 +8029,97 @@ void wm_vuldel_truncate_revision(char * revision) {
     return;
 }
 
+int wm_vuldet_recv_wdb_by_row(cJSON** j_array) {
+    char* response = NULL;
+    char* payload = NULL;
+    const char* error = NULL;
+    int size;
+    int parse_result;
+    int result = OS_INVALID;
+    bool sync_status = FALSE;
+    cJSON *j_object = NULL;
+    cJSON* j_status = NULL;
+
+    os_calloc(OS_MAXSTR, sizeof(char), response);
+    *j_array = cJSON_CreateArray();
+
+    do {
+        size = wm_vuldet_recv_wdb(response, OS_MAXSTR);
+        parse_result = wdbc_parse_result(response, &payload);
+        j_object = cJSON_ParseWithOpts(payload, &error, TRUE);
+        if (!j_object) {
+            mdebug1("Invalid JSON syntax in vulnerability detector database request.");
+            mdebug2("JSON error near: %s", error);
+            result = OS_INVALID;
+            break;
+        }
+
+        switch (parse_result) {
+            case WDBC_OK:
+                j_status = cJSON_GetObjectItem(j_object, "status");
+                if (cJSON_IsString(j_status)) {
+                    if (!strcmp(cJSON_GetStringValue(j_status),"SUCCESS")) {
+                        sync_status = TRUE;
+                        result = OS_SUCCESS;
+                    } else if ( !strcmp(cJSON_GetStringValue(j_status),"NOT_SYNCED")) {
+                        sync_status = FALSE;
+                        result = OS_SUCCESS;
+                    } else {
+                        sync_status = FALSE;
+                        result = OS_INVALID;
+                    }
+                } else {
+                    sync_status = FALSE;
+                    result = OS_INVALID;
+                }
+                break;
+            case WDBC_DUE:
+                cJSON_AddItemToArray(*j_array, j_object);
+                break;
+            default:
+                result = OS_INVALID;
+        }
+    } while (size != OS_INVALID && parse_result == WDBC_DUE);
+
+    if (OS_INVALID == result || FALSE == sync_status) {
+        cJSON_Delete(*j_array);
+    }
+    os_free(response);
+    return result;
+}
+
+int wm_vuldet_get_hotfixes(int agent_id, cJSON** requested_items) {
```

Comment:
You should add the definition of this function at the beginning of the file to allow usage.

Reply:
Definition and doxygen added !